### PR TITLE
Move the finalizer into a new internal package

### PR DESCRIPTION
**Pull request by qbarrand**
Make the code that deals with `SpecialResource` finalizations a distinct type so it's easier to unit test.
Move the new type to an `internal` package.

/cc @yevgeny-shnaidman

### CODE REVIEW

**Comment by qbarrand** [comment]
/ok-to-test

@ybettan I don't see tide in the list of checks... Could this be related to what you're working on?

**Comment by ybettan** [comment]
> I don't see tide in the list of checks... Could this be related to what you're working on?


@qbarrand I don't know why you didn't see it but I see it now.

**Comment by yevgeny-shnaidman** [comment]
@qbarrand why do we create an internal directory, and then the package inside that directory? Why can't it be just another package under pkg/ ?

**Comment by qbarrand** [comment]
@ybettan now it's there indeed... Not sure what happened.

@yevgeny-shnaidman Putting this into `internal` makes the code unimportable by any other project. It makes sure that the finalizer is not part of SRO's public API.

File: internal/controllers/finalizers/specialresource.go [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+	}
+}
+
+func (srf *specialResourceFinalizer) AddToSpecialResource(ctx context.Context, sr *v1beta1.SpecialResource) error {
+	srf.log.Info("Adding finalizer to special resource")
+	controllerutil.AddFinalizer(sr, FinalizerString)
+
+	// Update CR
+	if err := srf.kubeClient.Update(ctx, sr); err != nil {
+		srf.log.Info("Adding finalizer failed", "error", fmt.Sprintf("%v", err))
```
Comment (pmtk) [inline internal/controllers/finalizers/specialresource.go]: Nit: I see this code only changed place, but maybe we could use `srf.log.Error(err, "msg")` instead?

Comment (qbarrand) [reply internal/controllers/finalizers/specialresource.go]: Absolutely. But this is probably done elsewhere as well, so let's replace all occurrences in a dedicated PR?

Comment (pmtk) [reply internal/controllers/finalizers/specialresource.go]: Great. No problem with doing it in another PR

**Pull request merged**